### PR TITLE
Develop final

### PR DESCRIPTION
**Pull request by rowem5**
1. Updated resnet_beast.py to properly load beast checkpoints
2. Deleted code that forced one validation set. This is important to remove in datasets that are scarce and the benefit of having extra training data exceeds the benefit of a validation set. 
3. Updated tests to reflect the above changes
4. Updates tests file structure to mirror our nomenclature update in the models folder
5. Added new test files for the backbone loading files.

### CODE REVIEW

File: tests/models/backbones/test_resnet.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        """Test backbone properties."""
+        backbone = ResNetBackbone(default_config)
+        assert backbone.hidden_size == 512
+        assert backbone.num_channels == 3
+        assert backbone.image_size == 224
+        assert backbone.patch_size == 32
+        assert backbone.backbone_name == 'resnet18'
+        assert backbone.backbone_type == 'resnet'
+
+    def test_hidden_size_mapping(self):
```
Comment (themattinthehatt) [inline tests/models/backbones/test_resnet.py:52]: can remove, all these scenarios are covered by `test_all_supported_variants` right?

File: tests/models/backbones/test_resnet.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        for name in ['resnet18', 'resnet34']:
+            backbone = ResNetBackbone({'backbone': name})
+            assert backbone.hidden_size == 512
+
+        # resnet50/101/152 -> 2048
+        for name in ['resnet50', 'resnet101', 'resnet152']:
+            backbone = ResNetBackbone({'backbone': name})
+            assert backbone.hidden_size == 2048
+
+    def test_forward_pass_shape(self, default_config):
```
Comment (themattinthehatt) [inline tests/models/backbones/test_resnet.py]: update this to include all backbones `for name in RESNET_HIDDEN_SIZES:`

File: tests/models/backbones/test_resnet.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+    def test_forward_channel_validation(self, default_config):
+        """Test that wrong input channels raises ValueError."""
+        backbone = ResNetBackbone(default_config)
+        x = torch.randn(2, 1, 224, 224)  # wrong channels
+
+        with pytest.raises(ValueError, match="Input has 1 channels"):
+            backbone(x)
+
+    def test_forward_resnet50(self):
```
Comment (themattinthehatt) [inline tests/models/backbones/test_resnet.py]: can delete, will be covered by the expanded forward pass test above

File: tests/models/backbones/test_resnet_beast.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    def test_properties(self, default_config):
+        """Test backbone properties."""
+        backbone = ResNetBeastBackbone(default_config)
+        assert backbone.hidden_size == 512
+        assert backbone.num_channels == 3
+        assert backbone.image_size == 224
+        assert backbone.patch_size == 32
+        assert backbone.backbone_type == 'resnet-beast'
+
+    def test_hidden_size_mapping(self):
```
Comment (themattinthehatt) [inline tests/models/backbones/test_resnet_beast.py]: can generalize this with `BEAST_RESNET_HIDDEN_SIZES` right?

File: tests/models/backbones/test_resnet_beast.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        """Test hidden size mapping for different variants."""
+        for name in ['resnet18', 'resnet34']:
+            backbone = ResNetBeastBackbone({'backbone': name})
+            assert backbone.hidden_size == 512
+
+        for name in ['resnet50']:
+            backbone = ResNetBeastBackbone({'backbone': name})
+            assert backbone.hidden_size == 2048
+
+    def test_forward_pass_shape(self, default_config):
```
Comment (themattinthehatt) [inline tests/models/backbones/test_resnet_beast.py]: same as above, this would be a nice one to loop over `BEAST_RESNET_HIDDEN_SIZES` for

**Review by themattinthehatt — CHANGES_REQUESTED** [review]

File: tests/models/backbones/test_resnet_beast.py (hunk 4 above)
Comment (rowem5) [reply tests/models/backbones/test_resnet_beast.py]: https://github.com/paninski-lab/lightning-action/pull/33/changes/6efac5e3132d0731a29eeb096838da4c3cc16ad2

File: tests/models/backbones/test_resnet_beast.py (hunk 5 above)
Comment (rowem5) [reply tests/models/backbones/test_resnet_beast.py]: https://github.com/paninski-lab/lightning-action/pull/33/changes/6efac5e3132d0731a29eeb096838da4c3cc16ad2

File: tests/models/backbones/test_resnet.py (hunk 1 above)
Comment (rowem5) [reply tests/models/backbones/test_resnet.py:52]: https://github.com/paninski-lab/lightning-action/pull/33/changes/86e98cb95def21a7b604f09f50183e3c17ec3865

File: tests/models/backbones/test_resnet.py (hunk 2 above)
Comment (rowem5) [reply tests/models/backbones/test_resnet.py]: https://github.com/paninski-lab/lightning-action/pull/33/changes/86e98cb95def21a7b604f09f50183e3c17ec3865

File: tests/models/backbones/test_resnet.py (hunk 3 above)
Comment (rowem5) [reply tests/models/backbones/test_resnet.py]: https://github.com/paninski-lab/lightning-action/pull/33/changes/86e98cb95def21a7b604f09f50183e3c17ec3865

**Review by themattinthehatt — APPROVED** [review]
